src/Led.cpp

```diff
@@ -190,31 +187,69 @@ bool Led_NeedsProgressRedraw(bool lastPlayState, bool lastLockState,
 	return false;
 }
 
+CRGB Led_DimColor(CRGB color, uint8_t brightness) {
+	float factorBrighness = (float)brightness / DIMMABLE_STATES;
+	color.red = static_cast<uint8_t>((color.red * factorBrighness) + 0.5f);
+	color.green = static_cast<uint8_t>((color.green * factorBrighness) + 0.5f);
+	color.blue = static_cast<uint8_t>((color.blue* factorBrighness) + 0.5f);
+	return color;
+}
+
+void Led_DrawIdleDots(CRGB* leds, uint8_t offset, CRGB::HTMLColorCode color) {
+	for (uint8_t i=0; i<NUM_LEDS_IDLE_DOTS; i++){
+		leds[(Led_Address(offset)+i*Led_IdleDotDistance)%NUM_LEDS] = color;
+	}
+}
+
+bool CheckForPowerButtonAnimation() {
+	bool powerAnimation = false;
+	if (gShutdownButton < (sizeof(gButtons) / sizeof(gButtons[0])) - 1) { // Only show animation, if CMD_SLEEPMODE was assigned to BUTTON_n_LONG + button is pressed
+		if (!gButtons[gShutdownButton].currentState && (millis() - gButtons[gShutdownButton].firstPressedTimestamp >= 150) && gButtonInitComplete) {
+			powerAnimation = true;
+		}
+	} else {
+		gShutdownButton = (sizeof(gButtons) / sizeof(gButtons[0])) - 1; // If CMD_SLEEPMODE was not assigned to an enabled button, dummy-button is used
+		if (!gButtons[gShutdownButton].currentState) {
+			gButtons[gShutdownButton].currentState = true;
+		}
+	}
+	return powerAnimation;
+}
+
 static void Led_Task(void *parameter) {
 	#ifdef NEOPIXEL_ENABLE
 		static uint8_t hlastVolume = AudioPlayer_GetCurrentVolume();
```

Comment:
Can you mark again what exactly you men? The link in this review seems to show something else :-)

Reply:
Basically all the local variable starting from line 218 up until 244 are marked static (and so put into the data segment instead of the task stack). 

Purely looking at the code, they do not need to be static, since they are declared outside of the endless loop of the RTOS task, so they retain their values as long as the task exists. 

The main question is, if there was another aspect behind forcing the variables onto the data segment. The task has a stack of 1512 bytes reserved from the heap, which seems pretty reasonable for such a task without many functions calls or local variable.

Reply:
Well I really don't want to touch this. Let's keep this for the next PR? Before changing this we should evaluate the used stack-size.

Reply:
> @biologist79, do you perhaps know the reason why these variable are static (and thus in the data segment instead of the stack of this task)?
> 
> My gut feeling is, that they do not have to be (expect for `CRGB leds[NUM_LEDS]`, which should be global).

I guess you're right :-)